### ISSUE TITLE
httpsclient.cpp build errors

### ISSUE THREAD

**Issue by ninelore** [brainboxdotcc/DPP#671]
**Git commit reference**
10.0.23 via AUR
10.0.23 via vcpkg

**Describe the bug**
Build errors. AUR Example:
```
[ 82%] Building CXX object library/CMakeFiles/dpp.dir/__/src/dpp/httpsclient.cpp.o
In file included from /home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/httpsclient.h:27,
                 from /home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:27:
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/sslclient.h:127:9: error: ‘uint64_t’ does not name a type
  127 |         uint64_t bytes_out;
      |         ^~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/sslclient.h:27:1: note: ‘uint64_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
   26 | #include <dpp/socket.h>
  +++ |+#include <cstdint>
   27 |
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/sslclient.h:132:9: error: ‘uint64_t’ does not name a type
  132 |         uint64_t bytes_in;
      |         ^~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/sslclient.h:132:9: note: ‘uint64_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/sslclient.h:160:9: error: ‘uint64_t’ does not name a type
  160 |         uint64_t get_bytes_out();
      |         ^~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/sslclient.h:160:9: note: ‘uint64_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/sslclient.h:166:9: error: ‘uint64_t’ does not name a type
  166 |         uint64_t get_bytes_in();
      |         ^~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/sslclient.h:166:9: note: ‘uint64_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/httpsclient.h:35:17: error: found ‘:’ in nested-name-specifier, expected ‘::’
   35 | enum http_state : uint8_t {
      |                 ^
      |                 ::
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/httpsclient.h:35:6: error: ‘http_state’ has not been declared
   35 | enum http_state : uint8_t {
      |      ^~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/httpsclient.h:35:27: error: expected unqualified-id before ‘{’ token
   35 | enum http_state : uint8_t {
      |                           ^
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/httpsclient.h:116:9: error: ‘uint16_t’ does not name a type
  116 |         uint16_t port;
      |         ^~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/httpsclient.h:28:1: note: ‘uint16_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
   27 | #include <dpp/sslclient.h>
  +++ |+#include <cstdint>
   28 |
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/httpsclient.h:128:9: error: ‘http_state’ does not name a type
  128 |         http_state state;
      |         ^~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/httpsclient.h:154:9: error: ‘uint64_t’ does not name a type
  154 |         uint64_t content_length;
      |         ^~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/httpsclient.h:154:9: note: ‘uint64_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/httpsclient.h:166:9: error: ‘uint16_t’ does not name a type
  166 |         uint16_t status;
      |         ^~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/httpsclient.h:166:9: note: ‘uint16_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/httpsclient.h:213:9: error: ‘http_state’ does not name a type
  213 |         http_state get_state();
      |         ^~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/httpsclient.h:237:51: error: ‘uint16_t’ has not been declared
  237 |         https_client(const std::string &hostname, uint16_t port = 443, const std::string &urlpath = "/", const std::string &verb = "GET", const std::string &req_body = "", const http_headers& extra_headers = {}, bool plaintext_connection = false, uint16_t request_timeout = 5);
      |                                                   ^~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/httpsclient.h:237:248: error: ‘uint16_t’ has not been declared
  237 | std::string &urlpath = "/", const std::string &verb = "GET", const std::string &req_body = "", const http_headers& extra_headers = {}, bool plaintext_connection = false, uint16_t request_timeout = 5);
      |                                                                                                                                                                           ^~~~~~~~

/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/httpsclient.h:300:9: error: ‘uint16_t’ does not name a type
  300 |         uint16_t get_status() const;
      |         ^~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/httpsclient.h:300:9: note: ‘uint16_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:34:1: error: no declaration matches ‘dpp::https_client::https_client(const std::string&, uint16_t, const std::string&, const std::string&, const std::string&, const dpp::http_headers&, bool, uint16_t)’
   34 | https_client::https_client(const std::string &hostname, uint16_t port,  const std::string &urlpath, const std::string &verb, const std::string &req_body, const http_headers& extra_headers, bool plaintext_connection, uint16_t request_timeout)
      | ^~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/httpsclient.h:123:18: note: candidates are: ‘dpp::https_client::https_client(const dpp::https_client&)’
  123 | class DPP_EXPORT https_client : public ssl_client
      |                  ^~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/httpsclient.h:237:9: note:                 ‘dpp::https_client::https_client(const std::string&, int, const std::string&, const std::string&, const std::string&, const dpp::http_headers&, bool, int)’
  237 |         https_client(const std::string &hostname, uint16_t port = 443, const std::string &urlpath = "/", const std::string &verb = "GET", const std::string &req_body = "", const http_headers& extra_headers = {}, bool plaintext_connection = false, uint16_t request_timeout = 5);
      |         ^~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/httpsclient.h:123:18: note: ‘class dpp::https_client’ defined here
  123 | class DPP_EXPORT https_client : public ssl_client
      |                  ^~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp: In member function ‘virtual void dpp::https_client::connect()’:
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:52:9: error: ‘state’ was not declared in this scope; did you mean ‘static’?
   52 |         state = HTTPS_HEADERS;
      |         ^~~~~
      |         static
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:52:17: error: ‘HTTPS_HEADERS’ was not declared in this scope
   52 |         state = HTTPS_HEADERS;
      |                 ^~~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp: In member function ‘virtual bool dpp::https_client::handle_buffer(std::string&)’:
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:133:25: error: ‘state’ was not declared in this scope; did you mean ‘static’?
  133 |                 switch (state) {
      |                         ^~~~~
      |                         static
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:134:30: error: ‘HTTPS_HEADERS’ was not declared in this scope
  134 |                         case HTTPS_HEADERS:
      |                              ^~~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:166:65: error: ‘content_length’ was not declared in this scope
  166 |                                                                 content_length = std::stoull(response_headers["content-length"]);
      |                                                                 ^~~~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:168:65: error: ‘content_length’ was not declared in this scope
  168 |                                                                 content_length = ULLONG_MAX;
      |                                                                 ^~~~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:179:81: error: ‘HTTPS_CHUNK_LEN’ was not declared in this scope
  179 |                                                                         state = HTTPS_CHUNK_LEN;
      |                                                                                 ^~~~~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:183:57: error: ‘status’ was not declared in this scope; did you mean ‘static’?
  183 |                                                         status = atoi(req_status[1].c_str());
      |                                                         ^~~~~~
      |                                                         static
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:184:112: error: ‘content_length’ was not declared in this scope
  184 |                                                         if (status == 204  || status < 200 || status == 304 || content_length == 0) {
      |                                                                                                                ^~~~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:187:73: error: ‘HTTPS_CONTENT’ was not declared in this scope
  187 |                                                                 state = HTTPS_CONTENT;
      |                                                                         ^~~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:204:30: error: ‘HTTPS_CHUNK_CONTENT’ was not declared in this scope
  204 |                         case HTTPS_CHUNK_CONTENT: {
      |                              ^~~~~~~~~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:213:49: error: ‘HTTPS_CHUNK_TRAILER’ was not declared in this scope
  213 |                                         state = HTTPS_CHUNK_TRAILER;
      |                                                 ^~~~~~~~~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:220:30: error: ‘HTTPS_CHUNK_LAST’ was not declared in this scope
  220 |                         case HTTPS_CHUNK_LAST:
      |                              ^~~~~~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:221:30: error: ‘HTTPS_CHUNK_TRAILER’ was not declared in this scope
  221 |                         case HTTPS_CHUNK_TRAILER:
      |                              ^~~~~~~~~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:224:57: error: ‘HTTPS_DONE’ was not declared in this scope
  224 |                                                 state = HTTPS_DONE;
      |                                                         ^~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:228:57: error: ‘HTTPS_CHUNK_LEN’ was not declared in this scope
  228 |                                                 state = HTTPS_CHUNK_LEN;
      |                                                         ^~~~~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:234:30: error: ‘HTTPS_CHUNK_LEN’ was not declared in this scope
  234 |                         case HTTPS_CHUNK_LEN:
      |                              ^~~~~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:255:30: error: ‘HTTPS_CONTENT’ was not declared in this scope
  255 |                         case HTTPS_CONTENT:
      |                              ^~~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:258:54: error: ‘content_length’ was not declared in this scope
  258 |                                 if (body.length() >= content_length) {
      |                                                      ^~~~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:259:49: error: ‘HTTPS_DONE’ was not declared in this scope
  259 |                                         state = HTTPS_DONE;
      |                                                 ^~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:264:30: error: ‘HTTPS_DONE’ was not declared in this scope
  264 |                         case HTTPS_DONE:
      |                              ^~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp: At global scope:
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:274:10: error: no declaration matches ‘uint16_t dpp::https_client::get_status() const’
  274 | uint16_t https_client::get_status() const {
      |          ^~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:274:10: note: no functions named ‘uint16_t dpp::https_client::get_status() const’
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/library/../include/dpp/httpsclient.h:123:18: note: ‘class dpp::https_client’ defined here
  123 | class DPP_EXPORT https_client : public ssl_client
      |                  ^~~~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:282:1: error: ‘http_state’ does not name a type
  282 | http_state https_client::get_state() {
      | ^~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp: In member function ‘virtual void dpp::https_client::one_second_timer()’:
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:287:78: error: ‘class dpp::https_client’ has no member named ‘state’
  287 |         if ((this->sfd == SOCKET_ERROR || time(nullptr) >= timeout) && this->state != HTTPS_DONE) {
      |                                                                              ^~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:287:87: error: ‘HTTPS_DONE’ was not declared in this scope
  287 |         if ((this->sfd == SOCKET_ERROR || time(nullptr) >= timeout) && this->state != HTTPS_DONE) {
      |                                                                                       ^~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp: In member function ‘virtual void dpp::https_client::close()’:
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:294:13: error: ‘state’ was not declared in this scope; did you mean ‘static’?
  294 |         if (state != HTTPS_DONE) {
      |             ^~~~~
      |             static
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:294:22: error: ‘HTTPS_DONE’ was not declared in this scope
  294 |         if (state != HTTPS_DONE) {
      |                      ^~~~~~~~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp: In static member function ‘static dpp::http_connect_info dpp::https_client::get_host_info(std::string)’:
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:301:56: error: too many initializers for ‘dpp::http_connect_info’
  301 |         http_connect_info hci = { false, "http", "", 80};
      |                                                        ^
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:303:21: error: ‘struct dpp::http_connect_info’ has no member named ‘port’
  303 |                 hci.port = 443;
      |                     ^~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:313:21: error: ‘struct dpp::http_connect_info’ has no member named ‘port’
  313 |                 hci.port = 443;
      |                     ^~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:318:21: error: ‘struct dpp::http_connect_info’ has no member named ‘port’
  318 |                 hci.port = atoi(url.substr(colon_pos + 1, url.length()).c_str());
      |                     ^~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:319:25: error: ‘struct dpp::http_connect_info’ has no member named ‘port’
  319 |                 if (hci.port == 0) {
      |                         ^~~~
/home/ninelore/.cache/paru/clone/dpp/src/DPP-10.0.23/src/dpp/httpsclient.cpp:320:29: error: ‘struct dpp::http_connect_info’ has no member named ‘port’
  320 |                         hci.port = 80;
      |                             ^~~~
At global scope:
cc1plus: note: unrecognized command-line option ‘-Wno-unused-private-field’ may have been intended to silence earlier diagnostics
make[2]: *** [library/CMakeFiles/dpp.dir/build.make:1490: library/CMakeFiles/dpp.dir/__/src/dpp/httpsclient.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:98: library/CMakeFiles/dpp.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
```

**To Reproduce**
Try to compile with AUR or vcpkg

**Expected behavior**
Successful Build 

**Screenshots**
n/a

**System Details:**
 - OS: Arch Linux Rolling
 - Discord Client used for testing: n/a

**Additional context**
n/a

**Comment by Warpten** [comment]
Hitting this on gcc-13 as well. Looks to me like the only relevant error is the lack of cstdint include in httpsclient and sslclient, everything else is the bad compiler parsing.

**Comment by Mishura4** [comment]
This was fixed in 852e84acd0287148076b4bbf9ed4f5550307fa1a, make sure to pull the newest master branch
